lib/crypto/tasks.py

```diff
@@ -0,0 +1,20 @@
+import logging
+
+from celeryutils import task
+
+from addons.models import Addon
+from lib.crypto.packaged import SigningError
+
+log = logging.getLogger('z.task')
+
+
+@task
+def sign_addons(addon_ids, **kw):
+    log.info('[%s] Signing addons.' % len(addon_ids))
+    addons = Addon.objects.filter(pk__in=addon_ids)
+    for addon in addons:
+        for version in addon.versions.all():
+            try:
+                version.sign_files()
+            except SigningError as e:
+                log.warning('Failed signing addon %s: %s.' % (addon.pk, e))
```

Comment:
This looks like it could be made a little more efficient by querying the versions directly and not having to make an `Addon` object just to call `.versions.all()`. Something like `for version in Version.objects.filter(addon__id__in=addon_ids)`.